src/components/dataList/dataListContainer/dataListContainer.tsx

```diff
@@ -0,0 +1,78 @@
+import classNames from 'classnames';
+import { Children, useEffect, useMemo, type ComponentProps, type ReactElement } from 'react';
+import { CardEmptyState } from '../../cards';
+import type { IEmptyStateProps } from '../../states';
+import { useDataListContext } from '../dataListContext';
+import { DataListContainerSkeletonLoader } from './dataListContainerSkeletonLoader';
+
+export interface IDataListContainerState
+    extends Pick<IEmptyStateProps, 'heading' | 'description' | 'primaryButton' | 'secondaryButton'> {}
+
+export interface IDataListContainerProps extends ComponentProps<'div'> {
+    /**
+     * Skeleton element displayed when the DataList container state is set to loading.
+     */
+    SkeletonElement?: React.FC;
+    /**
```

Comment:
I did set it to `ReactNode` initially, the issue was that I couldn't render the skeleton-element inside a `.map` function and pass the `key` property as:
```
{displayLoadingElements && loadingItems.map((_value, index) => <SkeletonLoader key={index} />)}
```

Reply:
Hmmm. You could do something like this perhaps if you don't mind wrapping it in a div. The check is because the default skeleton *is* an actual FC.
```jsx
 {displayLoadingElements &&
                loadingItems.map((value, index) =>
                    typeof SkeletonLoader === 'function' ? (
                        <SkeletonLoader key={index} />
                    ) : (
                        <div key={index}>{value}</div>
                    ),
                )}
```

Reply:
That's a valid alternative, I'd say that we can review this implementation to allow any ReactNode if we need to or someone requests it (and update the related tests as well). We will never pass a simple `<div />` element as skeleton loader for now and this enforces creating an actual component instead of passing a complex react-node as property